case-studies/higgs/dataframe/analyzers/APCHiggsTools.cc

```diff
@@ -0,0 +1,117 @@
+#include "APCHiggsTools.h"
+using namespace APCHiggsTools;
+
+ROOT::VecOps::RVec<edm4hep::ReconstructedParticleData>  APCHiggsTools::muon_quality_check(ROOT::VecOps::RVec<edm4hep::ReconstructedParticleData> in){
```

Comment:
muon_quality_check: this takes as input a collection of RecoParticles, and returns another collection. One may expect that it filters the input muons, keeping only those that pass some quality criteria. While what you do is: check that there is a mu+ and a mu-, if that is the case you return the input collection, otherwise you return an empty collection. Having a boolean function, or a filter that would process only events that fulfil this condition, would seem more appropriate. Clement has examples with filters I think. Otherwise, if you want to stick to this structure, I suggest you change the name at least.

Reply:
indeed, this is extremely confusing to return the same collection or an empty one depending if we have at least a mu+ and a mu-. I also suggest to change the name of the function and the return type to an integer (not a boolean as it can cause some issues in RDF)